Upgrade webpack to V4.

NDevox opened the pull request:
Included changes are:

  - Removing CommonsChunk plugin (deprecated) for the SplitChunks plugin
  - Renaming app.css to commons.css as a result of the above point
  - Replacing ExtractTextPlugin with MiniCSSExtractPLugin
  - Adding production/development modes.
  - Removing now default plugins (UglifyJS and NamedModules)

### Relevant Issue & Description

Fixes #230 

### Screenshots, if applicable

No change

### Requirements

Upgrade npm dependencies.

NDevox commented:
I find the app.css -> commons.css change a bit dodgy but not sure what can be done about it at the moment.

mattrasband commented:
Looks good 👍 , thank you for tackling this. Just a few minor CI issues that are just style, I think.

I may have to wait to merge it until I can manage server changes if necessary (I will be out tonight, so may not be until tomorrow or Saturday) - unless of course you have time/access

NDevox commented:
Dang thought I'd got all of them. Updated.

NDevox commented:
Got to love yarn giving me a near 3k line diff.

Commit:
Upgrade webpack to V4.

Included changes are:

  - Removing CommonsChunk plugin (deprecated) for the SplitChunks plugin
  - Renaming app.css to commons.css as a result of the above point
  - Replacing ExtractTextPlugin with MiniCSSExtractPLugin
  - Adding production/development modes.
  - Removing now default plugins (UglifyJS and NamedModules)

NDevox committed:
Merge branch 'master' into upgrade-webpack-4

NDevox committed:
Merge branch 'master' into upgrade-webpack-4

mattrasband commented:
Sorry I dropped the ball with this, if you can get it up to date and have a passing build, I'll get it in.

Follow up note for myself: Collapse the `webpack.config` files and pass the build flag since they collapsed those a little in webpack.
